python/tank/util/metrics.py

```diff
@@ -324,100 +327,196 @@ def _dispatch(self, metrics):
 ###############################################################################
 # ToolkitMetric classes and subclasses
 
-class ToolkitMetric(object):
-    """Simple class representing tk metric data."""
+class EventMetric(object):
+    """
+    Convenience class for creating a metric event to be logged on a Shotgun site.
+
+    Use this helper class to create a suitable metric structure that you can
+    then pass to the `tank.utils.metrics.log_metric_event` method.
+
+    The simplest usage of this class is simply to provide an event group and
+    event name to the constructor:
+
+    Optionally, you can add your own specific metrics by using the
+    `properties` parameter. The latter simply takes a standard
+    dictionary.
+
+    The class also define numerous standard definition.
+    We highly recommand usage of them. Below is a complete typical usage:
+
+    ```
+    metric = EventMetric(EventMetric.GROUP_APP,
+        "User Logged In",
+        properties={
+            EventMetric.KEY_ENGINE: "tk-maya",
+            EventMetric.KEY_ENGINE_VERSION: "v0.2.2",
+            EventMetric.KEY_HOST_APP: "Maya",
+            EventMetric.KEY_HOST_APP_VERSION: "2017",
+            EventMetric.KEY_APP: "tk-multi-publish2",
+            EventMetric.KEY_APP_VERSION: "v0.2.3",
+            "CustomBoolMetric": True,
+            "RenderJobsSumitted": 173,
+        }
+    )
+    log_event_metric(metric)
+    ```
 
-    def __init__(self, data):
-        """Initialize the object with a dictionary of metric data.
-        
-        :param dict data: A dictionary of metric data.
-        
-        """
-        self._data = data
+    TODO: How about event type which is defined in the 'Shotgun Event Taxonomy' document as event property???
+    """
 
-    def __str__(self):
-        """Readable str representation of the metric."""
-        return "%s: %s" % (self.__class__, self._data)
+    """
+    Use groups to distinguish logical parts of your application.
+    Event groups help with prioritizing key components of your application
+    to instrument and can assist with event naming.
+    """
 
-    @property
-    def data(self):
-        """The underlying data this metric represents."""
-        return self._data
+    GROUP_TRIAL_SIGNUP = "Trial Signup"
```

Comment:
We should only include the toolkit groups called out in the tk page of the spreadsheet. I think there are only 6:

App, Tasks, Media, Toolkit, Navigation, Projects